receiver/splunkhecrawreceiver/receiver.go

```diff
@@ -0,0 +1,250 @@
+// By convention the consumer of the received data is set when the receiver
+// instance is created.
+func (r *splunkReceiver) Start(_ context.Context, host component.Host) error {
+	var ln net.Listener
+	// set up the listener
+	ln, err := r.config.HTTPServerSettings.ToListener()
+	if err != nil {
+		return fmt.Errorf("failed to bind to address %s: %w", r.config.Endpoint, err)
+	}
+
+	mx := mux.NewRouter()
+	mx.NewRoute().HandlerFunc(r.handleReq)
+
+	r.server = r.config.HTTPServerSettings.ToServer(mx)
+
+	// TODO: Evaluate what properties should be configurable, for now
+	//		set some hard-coded values.
+	r.server.ReadHeaderTimeout = defaultServerTimeout
+	r.server.WriteTimeout = defaultServerTimeout
+	r.resourceCustomizer = r.createResourceCustomizer()
+
+	go func() {
+		if errHTTP := r.server.Serve(ln); errHTTP != http.ErrServerClosed {
+			host.ReportFatalError(errHTTP)
+		}
+	}()
+
+	return err
+}
+
+// Shutdown tells the receiver that should stop reception,
+// giving it a chance to perform any necessary clean-up.
+func (r *splunkReceiver) Shutdown(context.Context) error {
+	return r.server.Close()
+}
+
+func (r *splunkReceiver) handleReq(resp http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	ctx = r.obsrecv.StartLogsOp(ctx)
+
+	if req.Method != http.MethodPost {
+		r.failRequest(ctx, resp, http.StatusBadRequest, invalidMethodRespBody, 0, errInvalidMethod)
+		return
+	}
+
+	encoding := req.Header.Get(httpContentEncodingHeader)
+	if encoding != "" && encoding != gzipEncoding {
+		r.failRequest(ctx, resp, http.StatusUnsupportedMediaType, invalidEncodingRespBody, 0, errInvalidEncoding)
+		return
+	}
+
+	if req.ContentLength == 0 {
+		resp.Write(okRespBody)
+		return
+	}
+
+	bodyReader := req.Body
+	if encoding == gzipEncoding {
+		var err error
+		bodyReader, err = gzip.NewReader(bodyReader)
+		if err != nil {
+			r.failRequest(ctx, resp, http.StatusBadRequest, errGzipReaderRespBody, 0, err)
+			return
+		}
+	}
+
+	sc := bufio.NewScanner(bodyReader)
+
+	ld := pdata.NewLogs()
+	rl := ld.ResourceLogs().AppendEmpty()
+	r.resourceCustomizer(req, rl.Resource())
+	ill := rl.InstrumentationLibraryLogs().AppendEmpty()
+
+	for sc.Scan() {
+		logRecord := ill.Logs().AppendEmpty()
+		logLine := sc.Text()
+		logRecord.Body().SetStringVal(logLine)
+	}
+	decodeErr := r.logsConsumer.ConsumeLogs(ctx, ld)
+
+	if decodeErr != nil {
+		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, ill.Logs().Len(), decodeErr)
+	} else {
+		resp.WriteHeader(http.StatusAccepted)
+		resp.Write(okRespBody)
+		r.obsrecv.EndLogsOp(ctx, typeStr, ill.Logs().Len(), nil)
+	}
+}
+
+func (r *splunkReceiver) failRequest(
+	ctx context.Context,
+	resp http.ResponseWriter,
+	httpStatusCode int,
+	jsonResponse []byte,
+	numLogsReceived int,
+	err error,
+) {
+	resp.WriteHeader(httpStatusCode)
+	if len(jsonResponse) > 0 {
+		// The response needs to be written as a JSON string.
+		resp.Header().Add("Content-Type", "application/json")
```

Comment:
Does this not need to be on the happy path too?

Reply:
Since we're removing OK, we don't need that either.